### PR TITLE
MM-14145: The config store Set will now Save automatically

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
When UpdateConfig (and configStore.Set) is called in admin.go and
config.go, commonStore.Set now takes a store-specific persist function.
It uses that persist function to save the configuration automatically.

Removed: Now callers do not have to call configStore.Save or
app.PersistConfig, and those functions have been removed.

Possible downside: this means a "failed to persist config" error can now
be thrown during a app.UpdateConfig or commonStore.Set call. But
considering application code never really sets a config without saving
it (except in the test cases, which were testing that -- see below), it
seems fine to group these responsibilities.

Also removed: tests for 'set without save'. Since that can not happen
anymore, the tests are not needed.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-14145
#10338

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

File: config/file.go [hunk 1]
```diff
@@ -86,15 +86,16 @@ func resolveConfigFilePath(path string) (string, error) {
...
+// Set replaces the current configuration in its entirety and updates the backing store.
 func (fs *FileStore) Set(newCfg *model.Config) (*model.Config, error) {
 	return fs.commonStore.set(newCfg, func(cfg *model.Config) error {
 		if *fs.config.ClusterSettings.Enable && *fs.config.ClusterSettings.ReadOnlyConfig {
 			return ErrReadOnlyConfiguration
 		}
 
 		return nil
-	})
+	},
```
Comment (lieut-data) [inline config/file.go]: nit: can you join this line with the next? the indentation is awkward otherwise.

File: config/store.go [hunk 2]
```diff
@@ -26,9 +26,6 @@ type Store interface {
 	// Load updates the current configuration from the backing store, possibly initializing.
 	Load() (err error)
 
-	// Save writes the current configuration to the backing store.
```
Comment (lieut-data) [inline config/store.go]: Can we now also eliminate the `Save` implementations on the concrete store instances?

File: config/database_test.go [hunk 3]
```diff
@@ -436,26 +436,13 @@ func TestDatabaseStoreSave(t *testing.T) {
 		},
 	}
 
-	t.Run("set without save", func(t *testing.T) {
+	t.Run("set with automatic save", func(t *testing.T) {
```
Comment (lieut-data) [inline config/database_test.go]: Let's remove `Test*StoreSave` altogether, and integrate this test case into `Test*StoreSet` instead.

File: config/store.go (hunk 2 above)
Comment (lieut-data) [reply config/store.go]: Also, can you update the comment on `Set` above in this interface to reflect the new functionality?

File: config/file.go (hunk 1 above)
Comment (cpoile) [reply config/file.go]: done!

File: config/store.go (hunk 2 above)
Comment (cpoile) [reply config/store.go]: Yes, we can eliminate all `Save` implementations, and I updated the comment on `Set`.

File: config/database_test.go (hunk 3 above)
Comment (cpoile) [reply config/database_test.go]: Done!

**Pull request merged**